### PR TITLE
Task 5 redux

### PR DESCRIPTION
**Pull request by hmaxim**

### CODE REVIEW

File: movie-app/src/containers/app/App.tsx [hunk 1]
```diff
@@ -1,60 +1,124 @@
...
+  const renderMovies = (movies: IMovie[], loading: boolean) => {
+    if (!loading) {
+      if (movies.length) {
+        return movies.map((item: IMovie) => {
+          return (
+            <MovieItem
+              navigate={() => navigate(item.id)}
+              {...item}
+              key={item.id + Math.random()}
+            ></MovieItem>
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/app/App.tsx:59]: don't use random as `keys`. `keys` should be predictable, so `id` is enough

File: movie-app/src/containers/search-container/SearchContainer.tsx [hunk 2]
```diff
@@ -1,28 +1,77 @@
...
     <SearchWrapper>
       <div className="search-container">
         <h1 className="search-title">FIND YOUR MOVIE</h1>
         <div>
-          <Input></Input>
-          <ButtonWrapper>SEARCH</ButtonWrapper>
-          <SwitchersButtons switchersTitle={'SEARCH BY'}
-          searchButtonsParams={searchButtonsParams}></SwitchersButtons>
+          <Input
+            onChange={event => onInputChange(event.target.value)}
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/search-container/SearchContainer.tsx]: Please, wrap with `useCallback`

File: movie-app/src/shared-components/switchers-buttons/SwitchersButtons.tsx [hunk 3]
```diff
@@ -10,9 +10,14 @@ const SwitchersButtons = (props: any) => {
...
       });
+
+      if (props.click && props.activeBtnValue !== arr[index].buttonValue) {
+        props.click(arr[index].buttonValue);
+      }
+
       updateSearchButtons(arr);
     },
-    [buttons]
+    [props.activeBtnValue],
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/switchers-buttons/SwitchersButtons.tsx:27]: `click` and `buttons` are missing at dependencies

File: movie-app/src/containers/list-header-container/ListHeaderContainer.tsx [hunk 4]
```diff
@@ -1,26 +1,76 @@
...
+import { useQuery, getNavUrl } from '../../constants/constants';
+import { useHistory, useLocation } from 'react-router';
 
 const ListHeaderContainer = (props: any) => {
+  const history = useHistory();
+  const location = useLocation();
+  const query = useQuery(location.search);
+  const isActive = (val: string): boolean => val === query.get('sortBy');
+
+  const sortButtons = [
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/list-header-container/ListHeaderContainer.tsx]: please, wrap with `useMemo`

File: movie-app/src/containers/list-header-container/ListHeaderContainer.tsx [hunk 5]
```diff
@@ -1,26 +1,76 @@
...
+import { setSearchParams } from '../../redux/rootActions';
+import { connect } from 'react-redux';
+import { useQuery, getNavUrl } from '../../constants/constants';
+import { useHistory, useLocation } from 'react-router';
 
 const ListHeaderContainer = (props: any) => {
+  const history = useHistory();
+  const location = useLocation();
+  const query = useQuery(location.search);
+  const isActive = (val: string): boolean => val === query.get('sortBy');
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/list-header-container/ListHeaderContainer.tsx]: please, wrap with `useCallback`

File: movie-app/src/containers/app/App.tsx [hunk 6]
```diff
@@ -1,60 +1,124 @@
...
     }
-    if (arr.length) {
-      return arr.map((item: IMovie) => {
-        return <MovieItem {...item} key={item.id + Math.random()}></MovieItem>;
-      });
-    } else {
-      return <EmptyState>No films found</EmptyState>;
+  }, [location]);
+
+  const navigate = (id: number) => {
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/app/App.tsx]: please. wrap with `useCallback`

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: movie-app/src/containers/app/App.tsx (hunk 1 above)
Comment (hmaxim) [reply movie-app/src/containers/app/App.tsx:59]: done

File: movie-app/src/containers/list-header-container/ListHeaderContainer.tsx (hunk 4 above)
Comment (hmaxim) [reply movie-app/src/containers/list-header-container/ListHeaderContainer.tsx]: Done

File: movie-app/src/containers/list-header-container/ListHeaderContainer.tsx (hunk 5 above)
Comment (hmaxim) [reply movie-app/src/containers/list-header-container/ListHeaderContainer.tsx]: Done

File: movie-app/src/shared-components/switchers-buttons/SwitchersButtons.tsx (hunk 3 above)
Comment (hmaxim) [reply movie-app/src/shared-components/switchers-buttons/SwitchersButtons.tsx:27]: Need to discuss

File: movie-app/src/containers/search-container/SearchContainer.tsx (hunk 2 above)
Comment (hmaxim) [reply movie-app/src/containers/search-container/SearchContainer.tsx]: Need to discuss

File: movie-app/src/containers/app/App.tsx (hunk 6 above)
Comment (hmaxim) [reply movie-app/src/containers/app/App.tsx]: Done

File: movie-app/src/constants/constants.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export const useQuery = (locationSearch: string) => {
```
Comment (AlexeyBarkow) [inline movie-app/src/constants/constants.ts:1]: it doesn't seem like a custom hook: this method doesn't  use any other hooks in the shell. I'd recommend to add here `useLocation` and `useMemo` here

File: movie-app/src/containers/app/App.tsx [hunk 8]
```diff
@@ -1,60 +1,126 @@
...
+  const history = useHistory();
+  const location = useLocation();
+  const query = useQuery(location.search);
 
-  const getMovies = () => {
-    const arr: IMovie[] = [];
-    for (let index = 0; index <= 10; index++) {
-      arr.push(MOVIE);
+  useEffect(() => {
+    if (!history.location.pathname.includes('film')) {
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/app/App.tsx]: why is this taken from history? why not from location?

File: movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx [hunk 9]
```diff
@@ -1,35 +1,68 @@
...
-const MovieDetailsContainer = (props: IMovie) => {
-  const genres = (genres: string[]) => <span>{genres.join(",")}</span>;
+const MovieDetailsContainer = (props: any) => {
+  const location = useLocation();
+  const params: any = useParams();
 
-  return (
-    <MovieDetailsWrapper score={props.vote_average}>
+  useEffect(() => {
+    if (props.history.location.pathname.includes('film')) {
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx]: same, why not use location from hook instead?

File: movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx [hunk 10]
```diff
@@ -1,35 +1,68 @@
...
-        {/* <img src={props.movie.poster_path} alt="poster" /> */}
+        <img src={props.selectedMovie.poster_path} alt="poster" />
       </div>
       <div className="movie-details">
         <div className="movie-title-container">
-          <h1 className="title">{props.title}</h1>
+          <h1 className="title">{props.selectedMovie.title}</h1>
           <div className="score">
-            {props.vote_average.toFixed(1)}
+            {+props.selectedMovie.vote_average.toFixed(1)}
```
Comment (AlexeyBarkow) [inline movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx]: `+` here seems to be redundant

File: movie-app/src/containers/app/App.tsx (hunk 8 above)
Comment (hmaxim) [reply movie-app/src/containers/app/App.tsx]: done

File: movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx (hunk 9 above)
Comment (hmaxim) [reply movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx]: done

File: movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx (hunk 10 above)
Comment (hmaxim) [reply movie-app/src/containers/movie-details-container/MovieDetailsContainer.tsx]: done

**Pull request merged**